regressors/include/iDynTree/Regressors/DynamicsRegressorGenerator.h

```diff
@@ -0,0 +1,312 @@
+     *     <subtreeBaseDynamics>
+     *         <FTSensorLink>l_ankle_2</FTSensorLink>
+     *         <FTSensorLink>l_hip_3</FTSensorLink>
+     *     </subtreeBaseDynamics>
+     * </regressor>
+     * ~~~
+     *
+     * For subtreeBaseDynamics subregressor, you specify the frame of expressions (i.e
+     * with respect to which point and to which orientation the regressor equation will be
+     * written) using the `<frame>` or `<sensorFrame>` tag.
+     *
+     * @param filename path to the file to load
+     * @param filetype type of the file to load, currently supporting only urdf type.
+     */
+    loadRegressorStructureFromFile(const std::string & filename);
+
+    /**
+     * Load the regressor structure from a configuration string, following the
+     * XML structure documented in loadRegressorStructureFromFile .
+     *
+     */
+    loadRegressorStructureFromString(const std::string & regressorStructureString);
+
+    /**
+     * Configure the regressor generator, i.e. check that the model and
+     * the regressor structure is compatible.
+     */
+    bool configure();
+
+    /**
+     * Return true if the configure call was successfull.
+     */
+    bool isValid();
+    //@}
+
+
+    /**
+     * @name Methods to get informations about the regressor
+     */
+    //@{
+    /**
+    *
+    * Get the number of parameters used by the regressor currently generated.
+    * This depends on the structure of the regressors added to the regressor generator.
+    * The number of the parameters will be the number of columns of the generated
+    * regressor.
+    *
+    * @return the number of parameters used by the regressor currently generated
+    */
+    int getNrOfParameters() const;
+
+    /**
+     * Get the number of outputs by the regressor currently generated.
+     * The number of outputs is the number of columns of the generated regressor.
+     *
+     */
+    int getNrOfOutputs() const;
+
+    /**
+     * Get a human readable description of a given parameter
+     */
+    std::string getDescriptionOfParameter(int parameter_index,
+                                          bool with_value=false,
+                                          double value=-1.0);
+
+    /**
+     * Get a human readable description of all the parameters
+     *
+     */
+    std::string getDescriptionOfParameters();
+
+
+    /**
+    * Get a human readable description of all the parameters.
+    *
+    * @return a std::string containg the description of the parameters
+    */
+    std::string getDescriptionOfParameters(const iDynTree::VectorDynSize & values);
+
+    /**
+     * Get a human readable description of a given output.
+     *
+     */
+    std::string getDescriptionOfOutput(int output_index);
+
+    /**
+     * Get a human readable description of all the outputs.
+     *
+     */
+    std::string getDescriptionOfOutputs();
+
+    /**
+     * Get a human readable description of a given inertal degree of freedom of the robot.
```

Comment:
Actually it was "internal" (i.e. not including the "floating base" degrees of freedom, but fixed.